### DISCUSSION TITLE
How to retrieve only local users and avoid querying LDAP users with get /users API in keycloak 22+ versions

### DISCUSSION THREAD

**Discussion by ashishkolte** [keycloak/keycloak#28313] [Q&A]
Hello,

We are currently trying to migrate from Keycloak 21 to 23 version. When we configure LDAP on Keycloak, in the 21 and below versions 
when we call the get all users API- /{realm}/users it retrieved only the local users which is what we required in our case. However, in the 22 and above versions we observed that it is also retrieving LDAP users and giving 400 error as there are a few users in LDAP that do not have username:

**Caused by: org.keycloak.models.ModelException: User returned from LDAP has null username! Check configuration of your LDAP mappings**

Is there a way we can avoid querying on LDAP and only return local users in the latest versions?

**Comment by pravsjv** [comment]
May be disabling LDAP integration and try fetching local users and then re-enable it if required.

**Reply by laptevn** [reply]
Quite painful option for me. What if automated process fails and doesn't enable LDAP afterwards? I need a manual intervention, quite complicated.
Is it possible to disable LDAP involvement in such user queries on Keycloak side?

LDAP participation makes the process slower in 10 times and leads to Hibernate connection closed errors. Somehow Keycloak keep DB connections alive while quering LDAP

**Reply by navatweb** [reply]
We are facing a similar issue in 26. When we query keycloak to check if a local user exists in keycloak, it appears it is going to ldap and timing out 
Is there a work around ?

**Reply by shawkins** [reply]
Please create an issue referencing this discussion. I don't see a work around for /users. Similar to how /users/id is able to use the id to determine what provider to query, /users would need to add some control which which providers to search.
